src/custom-elements/src/components/number-input/number-input.tsx

```diff
@@ -0,0 +1,182 @@
+import { Component, Prop, EventEmitter, Event } from '@stencil/core'
+import { CallbackQueue } from '../../utils/CallbackQueue'
+
+@Component({
+  tag: 'chi-number-input',
+  styleUrl: 'number-input.scss',
+  scoped: true
+})
+export class NumberInput {
+
+  private _didUpdateCallBackOnceQueue: (() => void)[] = [];
+
+  private initialValue;
+
+  /**
+   * used to hold the value of the number input
+   */
+  @Prop({ mutable: true, reflectToAttr: true }) value: string = '0';
+
+  /**
+   * used to set a step to indicate the expected granularity
+   */
+  @Prop({ reflectToAttr: true }) step: number = 1;
+
+  /**
+   * used to set a maximum allowed value
+   */
+  @Prop({ reflectToAttr: true }) max: number = Number.MAX_SAFE_INTEGER;
+
+  /**
+   * used to set a minimum allowed value
+   */
+  @Prop({ reflectToAttr: true }) min: number = Number.MIN_SAFE_INTEGER;
+
+  /**
+   * used to determine if component is disabled or not
+   */
+  @Prop({ reflectToAttr: true }) disabled?: boolean;
+
+  /**
+   * used to determine component's size
+   */
+  @Prop({ reflectToAttr: true }) size?: string;
+
+  /**
+   * used to set component's expanded mode
+   */
+  @Prop({ reflectToAttr: true }) expanded?: boolean;
+
+  /**
+   * used to determine if expanded component has a pill look
+   */
+  @Prop({ reflectToAttr: true }) pill?: boolean;
+
+  /**
+   * used to provide an input style like 'danger'. Mostly used for testing purposes
+   */
+  @Prop() inputstyle?: string;
+
+  /**
+   * used to provide an input state like 'hover' or 'focus'. Mostly used for testing purposes
+   */
+  @Prop() state?: string;
+
+  @Event() chiChange: EventEmitter<string>;
```

Comment:
can we emit a number instead of string?

Reply:
@srinath-koneru we are supporting strings just like in the standard component because the empty string must be an allowed value.

Reply:
@SergioQCL we don't resolve empty string to be 0 ?

Reply:
@srinath-koneru , we cannot resolve it to 0. If we where using flux stores (like redux) it will automatically put a 0 in the input.

Reply:
I don't think that's a good practice, because 0 is not equivalent to the empty string. The user could delete the value and leave it empty, and that wouldn't be 0 (although it behaved like that if increment or decrement buttons were pressed). But specially, I would't set that equivalence because that's not the standard component's behavior.